tests/TestCompressorDecompressor.cpp

```diff
@@ -0,0 +1,92 @@
+#include <external/catch2/Catch2.h>
+#include <external/expected/Expected.h>
+#include <internal/storage/Compressor.h>
+#include <internal/storage/Decompressor.h>
+
+bool create_testing_directory(const std::string &new_structure)
+{
+        for (int i = 0; i < 3; ++i) {
+                std::string filesystem_structure = new_structure;
+                if (i > 0)
+                        filesystem_structure += "/" + std::to_string(i);
+                
+                fs::path path{filesystem_structure};
+                path /= "file.txt";
+                if (!fs::create_directories(path.parent_path()))
+                        return false;
+                
+                std::ofstream ofs(path);
+                for (int j = 0; j < 200; ++j)
+                        ofs << "this is some text in the new file\n";
+                ofs.close();
+        }
+        return true;
+}
+
+bool exists(const fs::path &p, fs::file_status s = fs::file_status{ })
+{
+        if (fs::status_known(s) ? fs::exists(s) : fs::exists(p))
+                return true;
+        else
+                return false;
+}
+
+bool compare_folders(const std::string &first, const std::string &second)
+{
+        auto initial = fs::recursive_directory_iterator(first);
+        for (auto &compressed: fs::recursive_directory_iterator(second)) {
+                using compr = compression::storage::detail::CompressorInternal;
+                std::string compressed_np = compressed.path();
+                std::string initial_np = compressed.path();
+                
+                if (compressed_np != initial_np)
+                        return false;
+                
+                if (!compressed.is_directory() && !initial->is_directory()) {
+                        if (compr::return_file_info(compressed_np) != compr::return_file_info(initial_np))
+                                return false;
+                }
+                initial++;
+        }
+        return true;
+}
+
+bool clean(const std::map<std::string, std::string> &files)
+{
+        return std::ranges::all_of(files.cbegin(), files.cend(),
+                                   [ ](const std::pair<std::string, std::string> &i) {
+                                           return fs::remove_all(i.second);
+                                   });
+}
+
```

Comment:
I think we need a test that may help us define the limits of the current implementation. It may be enough to check whether any data is lost when compressing and decompressing really large data, say 100G.

Reply:
Resolved